File: Module1/SingletonPattern/Logger.java
public class Logger {

    private static Logger instance = new Logger();

    private Logger() {
        System.out.println("Logger Initialized");
    }

    public static Logger getInstance() {
        return instance;
    }

    public void log(String message) {
        System.out.println("Log: " + message);
    }
}

File: Module1/Singleton_C_Sharp/FirstProject/Logger.cs
using System;

public class Logger
{
    private static readonly Logger instance = new Logger();

    private Logger()
    {
        Console.WriteLine("Logger Initialized");
    }

    public static Logger GetInstance()
    {
        return instance;
    }

    public void Log(string message)
    {
        Console.WriteLine("Log: " + message);
    }
}
